### ISSUE TITLE
Newly added file is recognized as MiscellaneousFiles

### ISSUE THREAD

**Issue by arkalyanms** [microsoft/vscode-dotnettools#433]
From vscode-csharp created by [Cosifne](https://github.com/Cosifne): dotnet/vscode-csharp#6207  

Original comment: https://github.com/dotnet/vscode-csharp/issues/6023#issuecomment-1687626335
From the log it looks like part of the code is found in the project system, and the new file is not

**Comment by lifengl** [comment]
yes, using file system watcher to update project state doesn't work on Linux earlier. It should be fixed in the next pre-release.
